src/sentry/api/serializers/rest_framework/rule.py

```diff
@@ -184,13 +188,41 @@ def save(self, rule):
         return rule
 
 
+ACTION_UUID_KEY = "uuid"
+
+
+def _ensure_action_uuid(action: dict[Any, Any]) -> None:
+    """
+    Ensure that each action is uniquely identifiable.
+    The function will check that a UUID key and value exists in the action.
+    If the key does not exist, or it's not a valid UUID, it will create a new one and assign it to the action.
+
+    Does not add an uuid to the action if it is empty.
+    """
+    if not action:
+        return
+
+    if ACTION_UUID_KEY in action:
+        existing_uuid = action[ACTION_UUID_KEY]
+        try:
+            uuid.UUID(existing_uuid)
```

Comment:
Did you add this to only ensure it's not empty? What are the other invalid cases you're checking here? For example this would still pass if something is `uuid5` but not the format you expect.

Reply:
Yes! And this is to ensure it's a valid uuid, does not impose other restrictions or validity checks